nautobot/ipam/tests/test_api.py

```diff
@@ -102,6 +103,74 @@ def setUpTestData(cls):
         }
 
 
+class VRFDeviceAssignmentTest(APIViewTestCases.APIViewTestCase):
+    model = VRFDeviceAssignment
+
+    @classmethod
+    def setUpTestData(cls):
+        vrfs = VRF.objects.all()
+        devices = Device.objects.all()
+        locations = Location.objects.filter(location_type__name="Campus")
+        cluster_type = ClusterType.objects.create(name="Test Cluster Type")
+        clusters = (
+            Cluster.objects.create(name="Cluster 1", cluster_type=cluster_type, location=locations[0]),
+            Cluster.objects.create(name="Cluster 2", cluster_type=cluster_type, location=locations[1]),
+            Cluster.objects.create(name="Cluster 3", cluster_type=cluster_type, location=locations[2]),
+        )
+        vm_status = Status.objects.get_for_model(VirtualMachine).first()
+        vm_role = Role.objects.get_for_model(VirtualMachine).first()
+
+        test_vm = VirtualMachine.objects.create(
+            cluster=clusters[0],
+            name="VM 1",
+            role=vm_role,
+            status=vm_status,
+        )
+        VRFDeviceAssignment.objects.create(
```

Comment:
Would it make sense to create a `VRFDeviceAssignment` factory?

Reply:
We do have `DeviceFactory` to auto-generate device data. However, `ClusterType`/`ClusterGroup`/`VirtualMachine` factories are not enabled because that would require many updates https://github.com/nautobot/nautobot/blob/62f6195057e170a61f275e879e754c6165490ea9/nautobot/core/management/commands/generate_test_data.py#L213. Ideally we would enable all of them to make `VRFDeviceAssignmentFactory` but that would blow this story out of scope in my opinion.

Reply:
https://github.com/nautobot/nautobot/issues/5290